### PR TITLE
load speaker ids from external json if exist, and write default to json if not

### PR DESCRIPTION
**Pull request by greenwyrt**
the external json is called "spkr.json", and resides in CONFIG_DIR
the default speaker dict will be written if that file does not exist or "spkr.json" will be loaded on startup
if a change to speaker id is made through meta commands, save() is called to save the modified dict to json

### CODE REVIEW

File: plover_speaker_id/lookup.py [hunk 1]
```diff
@@ -1,3 +1,8 @@
+from plover.oslayer.config import CONFIG_DIR
+import json
+import pathlib
+spkr_dict_path = pathlib.Path(CONFIG_DIR) / "spkr.json"
```
Comment (sammdot) [inline plover_speaker_id/lookup.py:4]: nit: Should be upper case?

File: plover_speaker_id/lookup.py [hunk 2]
```diff
@@ -12,17 +17,27 @@
 }
 
 
+
 class SpeakerTable:
   def __init__(self):
     # TODO: load speaker IDs from external file in Plover's config directory
```
Comment (sammdot) [inline plover_speaker_id/lookup.py]: This _is_ what you're doing, right? Why is this comment still here?

File: plover_speaker_id/lookup.py [hunk 3]
```diff
@@ -12,17 +17,27 @@
...
-    self._speakers = DEFAULT_SPEAKERS
+    if not spkr_dict_path.exists():
+        with open(spkr_dict_path, "w") as f:
+            json.dump(DEFAULT_SPEAKERS, f)
```
Comment (sammdot) [inline plover_speaker_id/lookup.py]:
```python
self.save()
```

File: plover_speaker_id/lookup.py [hunk 4]
```diff
@@ -12,17 +17,27 @@
...
+        self._speakers = DEFAULT_SPEAKERS
+    else:
+        with open(spkr_dict_path, "r") as f:
+            self._speakers = json.loads(f.read(), object_hook=lambda d: {int(k) if k.lstrip('-').isdigit() else k: v for k, v in d.items()})
```
Comment (sammdot) [inline plover_speaker_id/lookup.py:28]: Not sure what's going on here -- are the speaker numbers converted to strings in the JSON? And why check for negative numbers?

File: plover_speaker_id/lookup.py (hunk 2 above)
Comment (greenwyrt) [reply plover_speaker_id/lookup.py]: I forgot to remove it...

File: plover_speaker_id/lookup.py (hunk 4 above)
Comment (greenwyrt) [reply plover_speaker_id/lookup.py:28]: when I dumped the dict to file, the int keys turned to strings, and load back as strings with json.loads. So the hook turns the stringed ints back to true int keys

File: plover_speaker_id/lookup.py (hunk 3 above)
Comment (greenwyrt) [reply plover_speaker_id/lookup.py]: ooh ... yes, DRY

**Comment by sammdot** [comment]
Thank you! 😄

**Pull request merged**